tests/bluetooth/tester/src/audio/btp_cap.c

```diff
@@ -697,6 +697,11 @@ static uint8_t btp_cap_broadcast_source_release(const void *cmd, uint16_t cmd_le
 
 	LOG_DBG("");
 
+	/* If no source has been created yet, there is nothing to release */
+	if (source == NULL || source->cap_broadcast == NULL) {
+		return BTP_STATUS_SUCCESS;
+	}
```

Comment:
@sjanc I think this is a missing part of the existing BTP documentation. The documentation only contain information about _what_ the command and responses contain, but unlike e.g. the HCI specification, it does not describe the expected behavior of commands in error cases. 

For example it seems that BlueKitchen will return success when removing or stopping a non-existing source, where Zephyr (until this PR) would return an error. 

I think that if we want BTP to be a proper protocol, we need to start expanding with additional status codes as well as expected behavior, otherwise it's basically a guessing game for the DUT when implementing it if the autopts client expects undocumented behavior.